[BB-4649] fix: Duplicate Arabic month

shimulch opened the pull request:
## Description

Arabic months are being shown in duplicates. The issue is from moment js. This issue has been [reported](https://github.com/moment/moment/issues/4270) and [solved](https://github.com/moment/moment/pull/4359) back in 2017. But unfortunately, Open edX [depending](https://github.com/edx/edx-platform/blob/master/package.json#L42) on a version before that. 

**Before this fix**
<img width="354" alt="Screenshot 2021-08-18 at 3 59 23 PM" src="https://user-images.githubusercontent.com/1010244/129882938-1d2616ff-732b-48e3-8533-9c918ca7230c.png">

**After this fix**
<img width="341" alt="Screenshot 2021-08-18 at 4 11 43 PM" src="https://user-images.githubusercontent.com/1010244/129882969-66df7967-9fcb-49dd-806d-684c344cd509.png">

## Testing instructions

1. Clone this fix in your local devstack.
2. Run lms - `make dev.up.lms`.
3. Change language to Arabic from - `http://localhost:18000/update_lang/`.
4. Visit - `http://localhost:18000/courses`.
5. Make sure that Arabic months are correct in the course start date.

## Deadline

None

## Other information

**Reviewers**
- [ ] @alfredchavez

edx-webhook commented:
Thanks for the pull request, @shimulch! I've created [OSPR-5973](https://openedx.atlassian.net/browse/OSPR-5973) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.




<!-- comment:external_pr -->
<!-- jenkins ok to test -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

giovannicimolin commented:
@shimulch Let me know when we can add this to the OSPR list (or schedule a CC review).

natabene commented:
@shimulch Thank you for your contribution. Please let me know once this is ready for our review.

natabene commented:
@OmarIthawi FYI this PR in case you are interested.

sarina commented:
What's the changeset from 2.19 to 2.29? Are there other changes we should be apprised of?

alfredchavez commented:
@shimulch  👍🏽 Looks good to me
- [x] I tested the changes, on devstack updated the language to arabic `ar`, visited the courses list and verified that month is not duplicated
- [x] I read through the change.

shimulch commented:
@giovannicimolin This can be added to OSPR list now.

@natabene This is ready to be reviewed.


> What's the changeset from 2.19 to 2.29? Are there other changes we should be apprised of?

@sarina I've just bump up to the latest version. Do you think we should only upgrade to the version this issue has been fixed? That would be quite an old version.

OmarIthawi reviewed (approved):
Thanks @shimulch. I've checked the locale update and a relatively minor and good translation update. 

I didn't check the other changes that might have come with `moment.js`.

**Out of scope of this pull request:**

This fix is totally acceptable, it standarizes the month names only for Egypt locale which might be weird for learners from Iraq and Syria which uses a slightly different locale.

This should be resolved at moment js by introducing another locale to have country-specific date format which adds a really nice UX. I've provided this feedback on the original moment js issue: https://github.com/moment/moment/issues/4270#issuecomment-903099582

sarina commented:
@shimulch I'm hesitant to just "upgrade to the latest version" of a package when we're unaware of what other changes have been introduced. Plus your JS tests are failing. I'd like you to understand what changes this bump introduces and make sure the tests are passing.

shimulch commented:
@sarina Thanks for raising concern regarding the upgrade to the latest version. I did check the change log and nothing actually seem alerting to me. But I do think jumping over a lot of releases is not a good idea. I've reworked on the PR and downgraded to [2.20.1](https://github.com/moment/moment/blob/develop/CHANGELOG.md#2201-see-changelog). I don't see any breaking changes there.

Furthermore, I've also fixed the test case. @OmarIthawi can you review if [this spec change](https://github.com/edx/edx-platform/pull/28486/files#diff-2d1157ec77c1e8f1839ccc72fbdc8f391ab2eda430995b10a3a0823f061a4490L28) is correct?

OmarIthawi commented:
@shimulch the spec change looks good. Thanks!

edx-webhook commented:
@shimulch 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/edx-platform&entry.867055334=https://github.com/edx/edx-platform/pull/28486&entry.1484655318=https://github.com/shimulch&entry.752974735=2021-08-18+10:32&entry.1917517419=2021-08-25+17:11&entry.2133058324=Yes) so we can improve your experience in the future.

<!-- comment:end_survey -->

shimulch commented:
@alfredchavez created https://github.com/edx/edx-platform/pull/29208 to cherry-pick this to `lilac.master`.
